pygfx/objects/_base.py

```diff
@@ -87,6 +93,23 @@ def __init__(self):
         self.visible = True
         self.render_order = 0
 
+        # See if we reached max number of objects. If so, try cleanup and try again.
+        if len(_idmap) >= _idmax:
+            gc.collect()
+            if len(_idmap) >= _idmax:
+                raise RuntimeError("Max number of object reached")
+        # Set id
+        self._id = id(self) % _idmax
+        while self._id in _idmap.values():
+            self._id = (self._id + 1) % _idmax
```

Comment:
Isn't this potentially very slow?

Reply:
A global counter could overflow if objects are continuously being created - you'd want to be able to re-use id's. This kinda uses uid's, derived from the object `id()`, the loop *could* take a long time, but assuming that the id's are randomly distributed, the average number of iterations is just 2 if we're halfway to the max. Maybe make it actually random (uid's) to make this assumption more realistic :D

Reply:
Made them actually random. And also decrease the max number of objects to 75% of the max id to prevent the set from "choking".